File: app/assets/javascripts/models/lesson.js
Textbook.Lesson = DS.Model.extend({
  name: DS.attr('string'),
  body: DS.attr('string'),
  number: DS.attr('number'),
  public: DS.attr('boolean'),
  created_at: DS.attr('date'),
  updated_at: DS.attr('date'),
  deleted_at: DS.attr('date'),
  section: DS.belongsTo('Textbook.Section')
});

File: app/models/lesson.rb
class Lesson < ActiveRecord::Base
  extend FriendlyId
  friendly_id :name, :use => :slugged

  acts_as_paranoid

  default_scope order :number

  attr_accessible :name, :body, :section_id, :number, :public, :deleted_at

  validates :name, :presence => true, :uniqueness => true
  validates :body, :presence => true
  validates :number, :presence => true, :numericality => {:only_integer => true}
  validates :section, :presence => true

  belongs_to :section

  before_destroy :set_private

  def next
    Lesson.where('number > ?', number).first
  end

  def previous
    Lesson.where('number < ?', number).last
  end

  def next_lesson?
    self.next != nil
  end

  def previous_lesson?
    self.previous != nil
  end

  private

  def set_private
    update_attributes(:public => false)
  end
end
